### ISSUE TITLE
bad_alloc error doing whole-collection queries using RocksDB

### ISSUE THREAD

**Issue by ropeladder** [arangodb/arangodb#3889]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [X] 3.2.8

Mode:
- [X] Single-Server

Storage-Engine:
- [X] rocksdb

On this operating system:
- [ ] Linux 
  - [X] Ubuntu .deb

### this is an AQL-related issue:
I'm issuing AQL via:
- [X] web interface with this browser: FireFox   running on this OS: Linux Mint 18

I initially posted this over in #3655 but opened a new issue since I'm using a single server (not a cluster).
Linux Mint 18 (Ubuntu), 16gb ram.

This will crash it reliably for me:
```
FOR n IN 1..10000000
    INSERT {n, otherData: "test test test"} INTO TestNums
    
FOR n IN TestNums
    UPDATE n WITH {otherData: "test some more"} IN TestNums
```

I also get similar errors with large REPLACE statements, and even had it crash a few times when using `arangodump`.

I get two different errors, one like this:
```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
```
And one mentioning that V8 is out of memory (don't have the exact error available right now, sorry).

**Comment by ropeladder** [comment]
(still crashes in 3.2.9)

**Comment by ropeladder** [comment]
Another query that takes 60 seconds, and then if you try to run it again it crashes ArangoDB with a similar bad_alloc message: (using the TestNums collection created above)

```
LET randomDoc = (FOR doc IN TestNums SORT RAND() LIMIT 1 RETURN doc._id)
RETURN randomDoc
```

**Comment by ropeladder** [comment]
Still present in 3.3.1.

This is a pretty serious issue for me.

**Comment by lewisdiamond** [comment]
Arangodb 3.3.2, rocksdb, docker
```
for d in data_py
    for o in d.data.observations
        insert merge({apMac: d.data.apMac}, o) into observations
```
About 2M data_py, targetting ~60M inserts, crashed after ~16M records in ~1100s.B
```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
```

**Comment by lewisdiamond** [comment]
Just an update, 
if I dont configure the machine with this:
```
sudo sysctl -w "vm.max_map_count=512000
sudo bash -c "echo madvise > /sys/kernel/mm/transparent_hugepage/enabled"
sudo bash -c "echo madvise > /sys/kernel/mm/transparent_hugepage/defrag"
sudo bash -c "echo 2 > /proc/sys/vm/overcommit_memory"
sudo bash -c "echo 100 > /proc/sys/vm/overcommit_ratio"
```
It goes through all the data without crashing. Turned out to be 40M records

**Comment by vinaypyati** [comment]
Hi @lewisdiamond,

ArangoDB 3.4 RC1 has been released. You can find it [here ](https://www.arangodb.com/download-technical-preview/). Please try reproducing the crash and let us know your feedback. 

Thanks!

**Comment by maxkernbach** [comment]
Hi @lewisdiamond, @ropeladder,

did you already try 3.4 RC1?
FYI: The latest available 3.4 Release Candidate is RC2 and can be found in the [download section](https://www.arangodb.com/download-technical-preview/).

Your feedback is highly appreciated!

**Comment by vinaypyati** [comment]
Hi,

We haven't received feedback from you for a while, I am closing this issue for now.

Please consider testing this issue on 3.4.0 RC3. You can download it from the [download section](https://www.arangodb.com/download-technical-preview/). If it does not resolve your problem, please feel free to reopen the issue by adding a new comment.

Thanks!
